invenio_rdm_records/records/models.py

```diff
@@ -54,6 +62,11 @@ class RDMRecordMetadata(db.Model, RecordMetadataBase, ParentRecordMixin):
     media_bucket_id = db.Column(UUIDType, db.ForeignKey(Bucket.id))
     media_bucket = db.relationship(Bucket, foreign_keys=[media_bucket_id])
 
+    # The deletion status is stored in the model so that we can use it in SQL queries
+    deletion_status = db.Column(
+        db.String(1), nullable=False, default=RecordDeletionStatusEnum.ACTIVE.value
+    )
```

Comment:
The JSON querying in PostgreSQL needs to have indexes defined on the value you want to query on. In such case query performance is comparable to the column query. Without the db index, the query performance is horrible on JSON fields.

The column fields are think at the best still since when s**t hits the fan, you need to operate fast, and not have to remember how you query into a specific JSON field (at least I never remember the syntax exactly have to look it up :-).